tests/virt_autotest/vgpu.pm

```diff
@@ -0,0 +1,368 @@
+            return;
+        }
+    }
+    # Install vGPU manager
+    download_script(${driver_file}, script_url => ${dirver_url});
+    zypper_call "in kernel-default-devel gcc";
+    assert_script_run("modprobe vfio_pci_core") if is_sle('>=15-SP5');
+    assert_script_run("./${driver_file} -s");
+    record_info("vGPU manager is installed successfully.");
+
+    # Reboot host(it took 180+ seconds for ph052 to bootup with calltrace)
+    enter_cmd("grub2-once 0; reboot");
+    record_info("Host rebooting ...");
+    select_console 'sol', await_console => 0;
+    # Wait for reboot and show screenshots
+    foreach (1 .. 10) {
+        save_screenshot;
+        sleep 20;
+    }
+    assert_screen([qw(sol-console-wait-typing-ret linux-login text-login)], 120);
+    if (match_has_tag('sol-console-wait-typing-ret')) {
+        send_key 'ret';
+        assert_screen([qw(inux-login text-login)], 120);
+    }
+    record_info("Host rebooted done");
+    reset_consoles;
+    select_console('root-ssh');
+
+    # Verify vGPU manager installed successfully
+    assert_script_run("lspci -d 10de: -k");
+    assert_script_run("lsmod | grep nvidia");
+    assert_script_run("nvidia-smi");
+    script_run("ls -ld /sys/class/m*");
+}
+
+sub enable_sriov {
+    my $gpu = shift;
+    assert_script_run("/usr/lib/nvidia/sriov-manage -e $gpu");
+    script_run("dmesg | tail");
+    assert_script_run("ls -l /sys/bus/pci/devices/0000:$gpu/ | grep virtfn");
+    assert_script_run("lspci | grep NVIDIA");
+    record_info("SR-IOV enabled");
+}
+
+sub enable_mig_mode {
+    # Enable MIG mode
+    assert_script_run("nvidia-smi -mig 1");
+    # Verify MIG mode is enabled
+    assert_script_run("nvidia-smi -i 0 --query-gpu=pci.bus_id,mig.mode.current --format=csv | grep 'Enabled'");
+    record_info("Mig Mode is enalbed!");
+}
+
+sub create_gpu_instance {
+
+    # Create a GI by randomly picking up a supported type which has free instances by the GPU device model
+    # Only those profiles on which there are available instances are candidates
+    script_run("nvidia-smi mig -lgip");
+    my $avail_gi_profile_cmd = "nvidia-smi mig -lgip | grep MIG | grep -v '0/' | grep -v '+me'";
+    die "No available GI profiles!" unless script_run($avail_gi_profile_cmd) == 0;
+    my @gi_profile_ids = split '\n', script_output("$avail_gi_profile_cmd | awk '{ print \$5 }'");
+    my $gi_profile = $gi_profile_ids[int(rand($#gi_profile_ids + 1))];
+    my ($gi_id, $gpu_id);
+    if (script_output("nvidia-smi mig -cgi $gi_profile") =~ /Successfully created GPU instance ID\s+(\d*) on GPU\s+(\d*)/) {
+        ($gi_id, $gpu_id) = ($1, $2);
+        assert_script_run("nvidia-smi mig -lgi");
+    }
+    else {
+        die "Fail to create a GPU Instance!";
+    }
+    record_info("GI created", "GPU_ID: \$gpu_id, GI_ID: \$gi_id");
+
+    return ($gi_id, $gpu_id);
+}
+
+sub create_compute_instance {
+    my $gi_id = shift;
+
+    # Create a CI to fully use a GI
```

Comment:
CI are GI are widely used. I can understand what they are after reading code. But maybe better to comment at the beginning or first usage for their full name

Reply:
done

Reply:
added in line 57/61/63